pkg/standalone-utils/test/GaugeActions.test.ts

```diff
@@ -0,0 +1,659 @@
+        });
+
+        itDepositsWithEnoughTokens();
+      });
+
+      context('sender = relayer, recipient = relayer', () => {
+        sharedBeforeEach('', async () => {
+          await lpToken.connect(userSender).transfer(relayer.address, totalLpTokens);
+          tokenSender = relayer;
+          tokenRecipient = relayer;
+        });
+
+        itDepositsWithEnoughTokens();
+      });
+    });
+
+    function itDepositsWithEnoughTokens() {
+      context('when depositing some of the tokens', () => {
+        itDepositsWithRefsAndAmounts(totalLpTokens.div(3));
+      });
+
+      context('when depositing all of the available tokens', () => {
+        itDepositsWithRefsAndAmounts(totalLpTokens);
+      });
+
+      context('when depositing 0 tokens', () => {
+        itDepositsWithRefsAndAmounts(bn(0));
+      });
+    }
+
+    function itDepositsWithRefsAndAmounts(amount: BigNumber) {
+      context('when using immediate amounts', () => {
+        itDepositsTokens(amount, amount);
+      });
+
+      context('when using chained references', () => {
+        const reference = toChainedReference(123);
+
+        sharedBeforeEach('set chained reference', async () => {
+          await setChainedReferenceContents(relayer, reference, amount);
+        });
+
+        itDepositsTokens(reference, amount);
+      });
+    }
+
+    function itDepositsTokens(depositAmountOrRef: BigNumber, expectedAmount: BigNumber) {
+      let receipt: ContractReceipt;
+      let tokenSenderAddress: string;
+      let tokenRecipientAddress: string;
+
+      sharedBeforeEach('get addresses', async () => {
+        tokenSenderAddress = TypesConverter.toAddress(tokenSender);
+        tokenRecipientAddress = TypesConverter.toAddress(tokenRecipient);
+      });
+
+      sharedBeforeEach('check initial balances and make deposit', async () => {
+        // Start: sender has all the BPT, no gauge tokens minted, recipient is clean unless it is sender.
+        expect(await lpToken.balanceOf(gauge.address)).to.be.eq(0);
+        expect(await lpToken.balanceOf(tokenSenderAddress)).to.be.eq(totalLpTokens);
+        if (tokenSenderAddress != tokenRecipientAddress) {
+          expect(await lpToken.balanceOf(tokenRecipientAddress)).to.be.eq(0);
+        }
+
+        expect(await gauge.balanceOf(gauge.address)).to.be.eq(0);
+        expect(await gauge.balanceOf(tokenSenderAddress)).to.be.eq(0);
+        expect(await gauge.balanceOf(tokenRecipientAddress)).to.be.eq(0);
+
+        const tx = await relayer.connect(userSender).multicall([
+          encodeDeposit({
+            gauge: gauge,
+            sender: tokenSenderAddress,
+            recipient: tokenRecipientAddress,
+            amount: depositAmountOrRef,
+          }),
+        ]);
+        receipt = await tx.wait();
+      });
+
+      // Short-circuit when no tokens are deposited.
+      expectedAmount.gt(0) &&
+        it('pulls BPT tokens from sender if necessary', async () => {
```

Comment:
Do a standard `if` :joy:

Reply:
The difference with the `if` is that the test doesn't appear at all when running it, and I thought that was more accurate than passing an empty test :grimacing: 

I'll do the change anyways :)

Reply:
Ah no, I didn't mean skip the test body, but rather skip the `it` call:
```
if (expectedAmount.gt(0)) {
 it('pulls tokens....', ....);
}
```